src/ScriptCs.Core/ReferenceLineProcessor.cs

```diff
@@ -32,16 +32,27 @@ protected override BehaviorAfterCode BehaviorAfterCode
         protected override bool ProcessLine(IFileParser parser, FileParserContext context, string line)
         {
             Guard.AgainstNullArgument("context", context);
-
+            
             var argument = GetDirectiveArgument(line);
-            var assemblyPath = Environment.ExpandEnvironmentVariables(argument);
 
-            var referencePath = _fileSystem.GetFullPath(assemblyPath);
+
+            var expandedArgument = Environment.ExpandEnvironmentVariables(argument);
+
+            if (argument.ToLower().StartsWith(Constants.PaketPrefix))
+            {
+                if (!context.CustomReferences.Contains(argument))
```

Comment:
maybe consider trim on the argument, it avoids trailing space (on both ends) to mess with duplicate handling you have in place.